src/pypyjs.js

```diff
@@ -566,10 +582,27 @@ pypyjs.prototype.exec = function exec(code) {
       });
     }
 
+    // if any modules have been re-added then we need to remove them from
+    // sys.modules which clears them from memory and allows them to be reloaded
+    // from the emscripten file system
+    if (Object.keys(this._modulesToReset).length) {
+      // construct python code to remove module from sys.modules:
+      // ```python
+      //   import sys
+      //   if 'foo' in sys.modules: del(sys.modules['foo'])
+      // ```
+      preCode = 'try:\n  import sys\n';
+      for (let module of Object.keys(this._modulesToReset)) {
```

Comment:
Without trying to sounds like an ass :smile: could you write this as a `map` function?